lib/delivery-locations-resolver.js

```diff
@@ -0,0 +1,117 @@
+const SCSBRestClient = require('@nypl/scsb-rest-client')
+const deliveryLocationByRecapCustomerCode = require('@nypl/nypl-core-objects')('by-recap-customer-codes')
+const logger = require('./logger')
+
+function deliveryLocationsByHoldingLocation (holdingLocation) {
+  // Mocked based on actual mapping for holdingLocation 'loc:scff2'
+  return [
+    {
+      id: 'loc:sc',
+      label: 'Schomburg Center'
+    }
+  ].map((loc) => Object.assign(loc, { label: `${loc.label}${!holdingLocation || holdingLocation.id !== 'loc:scff2' ? ' (mocked)' : ''}` }))
+}
+
+function deliveryLocationsByCustomerCode (customerCode) {
+  if (deliveryLocationByRecapCustomerCode[customerCode] && deliveryLocationByRecapCustomerCode[customerCode].sierraDeliveryLocations) {
+    return deliveryLocationByRecapCustomerCode[customerCode].sierraDeliveryLocations.map((ent) => {
+      return {
+        id: `loc:${ent.code}`,
+        label: ent.label
+      }
+    })
+  } else {
+    // Mocked based on actual mapping (first three) for customerCode 'PA'
+    return [
+      {
+        id: 'loc:maf',
+        label: 'SASB - Dorot Jewish Division Rm 111'
+      },
+      {
+        id: 'loc:mar',
+        label: 'SASB - Rare Book Collection Rm 328'
+      },
+      {
+        id: 'loc:mao',
+        label: 'SASB - Manuscripts & Archives Rm 328'
+      }
+    ].map((loc) => Object.assign(loc, { label: `${loc.label}${customerCode !== 'PA' ? ' (mocked)' : ''}` }))
+  }
+}
+
+function recapCustomerCodesByBarcodes (barcodes) {
+  let scsbClient = new SCSBRestClient({ url: process.env.SCSB_URL, apiKey: process.env.SCSB_API_KEY })
+
+  // Record time to process all:
+  var __startAll = new Date()
+
+  return Promise.all(
+    barcodes.map((barcode) => {
+      // Record start time to process this request
+      var __start = new Date()
+      return scsbClient.searchByParam({ fieldValue: barcode, fieldName: 'Barcode' })
+        .then((response) => {
+          let ellapsed = ((new Date()) - __start)
+          logger.debug({ message: `HTC searchByParam API took ${ellapsed}ms`, metric: 'searchByParam-barcode', timeMs: ellapsed })
+
+          if (response && response.length > 0 && (typeof response[0]) === 'object') {
+            return { [barcode]: response[0].customerCode }
+          }
+        })
+        .catch((error) => {
+          // This is a common error:
+          //  "Error hitting SCSB API 502: <html>\r\n<head><title>502 Bad Gateway</title></head>\r\n<body bgcolor=\"white\">\r\n<center><h1>502 Bad Gateway</h1></center>\r\n</body>\r\n</html>\r\n"
+          // return Promise.reject(error)
+          logger.info({ message: 'HTC API error. Send everything to NH', htcError: error.message })
```

Comment:
This would be a good place to `logger.errror()` so we can build alerts off of it.